src/common/m_helper.f90

```diff
@@ -0,0 +1,191 @@
+
+    ! ==========================================================================
+
+    implicit none
+
+    real(kind(0d0)) :: cart_y, cart_z
+    real(kind(0d0)) :: sph_phi !<
+    !! Variables to be used to hold cell locations in Cartesian coordinates if
+    !! 3D simulation is using cylindrical coordinates
+
+    private; public :: s_compute_finite_difference_coefficients, &
+        s_comp_n_from_prim, &
+        s_comp_n_from_cons, &
+        s_quad
+
+contains
+
+#ifdef MFC_SIMULATION
+    !>  The purpose of this subroutine is to compute the finite-
+        !!      difference coefficients for the centered schemes utilized
+        !!      in computations of first order spatial derivatives in the
+        !!      s-coordinate direction. The s-coordinate direction refers
+        !!      to the x-, y- or z-coordinate direction, depending on the
+        !!      subroutine's inputs. Note that coefficients of up to 4th
+        !!      order accuracy are available.
+        !!  @param q Number of cells in the s-coordinate direction
+        !!  @param s_cc Locations of the cell-centers in the s-coordinate direction
+        !!  @param fd_coeff_s Finite-diff. coefficients in the s-coordinate direction
+    subroutine s_compute_finite_difference_coefficients(q, s_cc, fd_coeff_s, buff_size, &
+                                                                fd_number, fd_order)
+
+        integer, intent(IN) :: q
+        integer, intent(IN) :: buff_size, fd_number, fd_order
+
+        real(kind(0d0)), &
+            dimension(-buff_size:q + buff_size), &
+            intent(IN) :: s_cc
+
+        real(kind(0d0)), &
+            dimension(-fd_number:fd_number, 0:q), &
+            intent(INOUT) :: fd_coeff_s
+
+        integer :: i !< Generic loop iterator
+
+        ! Computing the 1st order finite-difference coefficients
+        if (fd_order == 1) then
+            do i = 0, q
+                fd_coeff_s(-1, i) = 0d0
+                fd_coeff_s(0, i) = -1d0/(s_cc(i + 1) - s_cc(i))
+                fd_coeff_s(1, i) = -fd_coeff_s(0, i)
+            end do
+
+            ! Computing the 2nd order finite-difference coefficients
+        elseif (fd_order == 2) then
+            do i = 0, q
+                fd_coeff_s(-1, i) = -1d0/(s_cc(i + 1) - s_cc(i - 1))
+                fd_coeff_s(0, i) = 0d0
+                fd_coeff_s(1, i) = -fd_coeff_s(-1, i)
+            end do
+
+            ! Computing the 4th order finite-difference coefficients
+        else
+            do i = 0, q
+                fd_coeff_s(-2, i) = 1d0/(s_cc(i - 2) - 8d0*s_cc(i - 1) - s_cc(i + 2) + 8d0*s_cc(i + 1))
+                fd_coeff_s(-1, i) = -8d0*fd_coeff_s(-2, i)
+                fd_coeff_s(0, i) = 0d0
+                fd_coeff_s(1, i) = -fd_coeff_s(-1, i)
+                fd_coeff_s(2, i) = -fd_coeff_s(-2, i)
+            end do
+
+        end if
+
+    end subroutine s_compute_finite_difference_coefficients ! --------------
+#else
```

Comment:
merge these two subroutines by placing an ifdef on the loop bounds, since that seems to be the only difference? or am I missing something?

Reply:
Fixed with an optional argument instead

Reply:
Figured that was slightly cleaner